workspaces/src/network/sandbox.rs

```diff
@@ -135,36 +139,261 @@ impl Sandbox {
         ImportContractTransaction::new(id.to_owned(), worker.client(), self.client())
     }
 
-    pub(crate) async fn patch_state(
+    pub(crate) fn patch_state(&self, account_id: AccountId) -> SandboxPatchStateBuilder {
+        SandboxPatchStateBuilder::new(self, account_id)
+    }
+
+    pub(crate) fn patch_account(&self, account_id: AccountId) -> SandboxPatchStateAccountBuilder {
+        SandboxPatchStateAccountBuilder::new(self, account_id)
+    }
+
+    pub(crate) fn patch_access_key(
         &self,
-        contract_id: &AccountId,
-        key: &[u8],
-        value: &[u8],
-    ) -> anyhow::Result<()> {
-        let state = StateRecord::Data {
-            account_id: contract_id.to_owned(),
-            data_key: key.to_vec(),
-            value: value.to_vec(),
+        account_id: AccountId,
+        public_key: crate::types::PublicKey,
+    ) -> SandboxPatchAcessKeyBuilder {
+        SandboxPatchAcessKeyBuilder::new(self, account_id, public_key)
+    }
+
+    // shall we expose convenience patch methods here for consistent API?
+
+    pub(crate) async fn fast_forward(&self, delta_height: u64) -> anyhow::Result<()> {
+        // NOTE: RpcSandboxFastForwardResponse is an empty struct with no fields, so don't do anything with it:
+        self.client()
+            // TODO: replace this with the `query` variant when RpcSandboxFastForwardRequest impls Debug
+            .query_nolog(&RpcSandboxFastForwardRequest { delta_height })
+            .await
+            .map_err(|err| anyhow::anyhow!("Failed to fast forward: {:?}", err))?;
+
+        Ok(())
+    }
+}
+
+//todo: review naming
+#[must_use = "don't forget to .apply() this `SandboxPatchStateBuilder`"]
+pub struct SandboxPatchStateBuilder<'s> {
+    sandbox: &'s Sandbox,
+    account_id: AccountId,
+    records: Vec<StateRecord>,
+}
+
+impl<'s> SandboxPatchStateBuilder<'s> {
+    pub fn new(sandbox: &'s Sandbox, account_id: AccountId) -> Self {
+        SandboxPatchStateBuilder {
+            sandbox,
+            account_id,
+            records: Vec::with_capacity(4),
+        }
+    }
+
+    pub fn data(mut self, key: impl Into<Vec<u8>>, value: impl Into<Vec<u8>>) -> Self {
+        let data = StateRecord::Data {
+            account_id: self.account_id.clone(),
+            data_key: key.into(),
+            value: value.into(),
         };
-        let records = vec![state];
 
+        self.records.push(data);
+        self
+    }
+
+    pub fn data_multiple(
+        mut self,
+        kvs: impl IntoIterator<Item = (impl Into<Vec<u8>>, impl Into<Vec<u8>>)>,
+    ) -> Self {
+        let Self {
+            ref mut records,
+            ref account_id,
+            ..
+        } = self;
+        records.extend(kvs.into_iter().map(|(key, value)| StateRecord::Data {
+            account_id: account_id.clone(),
+            data_key: key.into(),
+            value: value.into(),
+        }));
```

Comment:
Minor: there's logical duplication between the two methods, one could delegate to the other (or both could delegate to internal _state_record)